### DISCUSSION TITLE
Can hx-trigger="load" only execute if the element is visible?

### DISCUSSION THREAD

**Discussion by deanebarker** [bigskysoftware/htmx#1142] [General]
I have some elements that are hidden on mobile. They have `hx-trigger="load"` on them, and the `GET` request is executing even when they're not shown.

I tried changing it to `hx-trigger="revealed"`, but they still executed, because they are technically in the viewport, they're just not visible.

I'd like this to _not_ execute if they're hidden.

**Comment by davidjr82** [comment]
Same issue here, did you find a solution?

**Comment by deanebarker** [comment]
No, sorry. But I did figure out how to tie stuff to document events, which is probably how I would approach it now.

https://stackoverflow.com/questions/74933490/how-do-i-trigger-an-element-based-htmx-get-request-from-vanilla-javascript

(I answered my own question there...)

**Reply by davidjr82** [reply]
But this approach would not cover when the element is/becomes visible, only when the page itself goes visible, as from a tab change or whatever. I have created a proposal issue, just in case this topic is still relevant for you #1527

**Comment by deanebarker** [comment]
Yes, in that example, I was listening for an event on the body (so, the entire document). I believe you can listen for events on specific elements too.
